### PR TITLE
Add Cart Rule product subselect total incl. tax option

### LINKED ISSUE #34871: Cart Rule product subselect condition is missing total including tax option
### Preconditions (*)

1. 2.4.3-p1

### Steps to reproduce (*)

1. Create a Cart price rule and add a products subselection condition. There is only a total amount option limiting the condition to be set only for a value excluding tax.

### Expected result (*)

1. Have the option to add the the total excluding tax as a condition option in products subselection

### Actual result (*)

1. There is only one option to select total amount excluding tax in the product subselection condition

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [x] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by johnulecan**
### Description (*)

Add Cart Rule product subselect total incl. tax option

### Fixed Issues (if relevant)

1. Fixes magento/magento2#34871

### Manual testing scenarios (*)

1. Create a new cart rule
2. Add a product subselect condition
3. Select `total (incl.tax)` option
4. Add a value including tax to the rule condition
5. Test if the rule is applied to subselection total including tax

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by johnulecan** [comment]
@magento give me test instance

**Comment by johnulecan** [comment]
@magento run all tests

**Comment by johnulecan** [comment]
@magento run all tests

**Comment by johnulecan** [comment]
@magento run WebAPI Tests, Functional Tests EE, Functional Tests B2B

**Comment by johnulecan** [comment]
@magento run Functional Tests EE

**Comment by johnulecan** [comment]
@magento run Functional Tests EE

File: app/code/Magento/SalesRule/Model/Rule/Condition/Product/Subselect.php [hunk 1]
```diff
@@ -68,7 +68,13 @@ public function asXml($containerKey = 'conditions', $itemKey = 'condition')
      */
     public function loadAttributeOptions()
     {
-        $this->setAttributeOption(['qty' => __('total quantity'), 'base_row_total' => __('total amount')]);
+        $this->setAttributeOption(
+            [
+                'qty' => __('total quantity'),
+                'base_row_total' => __('total amount (excl. tax)'),
+                'row_total_incl_tax' => __('total amount (incl. tax)')
```
Comment (Den4ik) [inline app/code/Magento/SalesRule/Model/Rule/Condition/Product/Subselect.php]: I believe that we should use `base_row_total_incl_tax` here

**Review by Den4ik — CHANGES_REQUESTED** [review]
Hi @johnulecan
Thanks for your PR.
Please look at my comment below

**Comment by Den4ik** [comment]
Could you also add some tests for add case

File: app/code/Magento/SalesRule/Model/Rule/Condition/Product/Subselect.php (hunk 1 above)
Comment (johnulecan) [reply app/code/Magento/SalesRule/Model/Rule/Condition/Product/Subselect.php]: Hey @Den4ik thank you for the feedback. I've modified the suggested line.

**Comment by johnulecan** [comment]
@magento run all tests

**Comment by johnulecan** [comment]
@magento run Functional Tests B2B

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Comment by engcom-Hotel** [comment]
@magento run Database Compare, Functional Tests CE, Functional Tests EE

**Review by engcom-Hotel — CHANGES_REQUESTED** [review]
Hello @johnulecan,

Thanks for the contribution!

Please sign CLA in order to proceed further with this PR.

Thanks

**Comment by johnulecan** [comment]
@magento run all tests

**Comment by johnulecan** [comment]
hey @engcom-Hotel I had the CLA signed, I re signed it now. Thanks.

**Comment by johnulecan** [comment]
@magento run WebAPI Tests, Functional Tests EE, Functional Tests B2B

**Comment by johnulecan** [comment]
@magento run all tests

**Comment by johnulecan** [comment]
@magento run Unit Tests, Functional Tests CE, Functional Tests B2B

**Comment by johnulecan** [comment]
@magento run Functional Tests B2B

**Review by engcom-Hotel — CHANGES_REQUESTED** [review]
Hello @johnulecan,

Thanks for the quick response!

I would like to suggest adding some automated tests to cover the code changes.

Thanks

**Comment by johnulecan** [comment]
@magento run all tests

**Comment by johnulecan** [comment]
https://github.com/magento run all tests

**Comment by johnulecan** [comment]
@magento run all tests

**Comment by johnulecan** [comment]
@magento run WebAPI Tests, Functional Tests EE

**Comment by johnulecan** [comment]
@engcom-Hotel Hey. I added a unit test to validate the rule and re-run all tests. Thank you.

**Review by engcom-Hotel — APPROVED** [review]

**Comment by engcom-Lima** [comment]
@magento give me test instance

**Comment by engcom-Lima** [comment]
### :heavy_check_mark: QA Passed

**Preconditions:**
- Install fresh Magento **`2.4-develop`** and **`PHP 8.1`**
- Tax should be configured.

**Manual testing scenario:**
1. Create a new cart rule
2. Add a product subselect condition
3. Select the total (incl.tax) option
4. Add a value including tax to the rule condition
5. Test if the rule is applied to the subselection total including tax
6. Select total (excl.tax) option
7. Add a value including tax to the rule condition
8. Test if the rule is applied to the subselection total including tax

**Before:** :heavy_multiplication_x: 

Screenshot 2023-09-25 at 3 56 53 PM


**After:** :heavy_check_mark:

Screenshot 2023-09-25 at 3 52 59 PM

**Excluding Tax:**
Screenshot 2023-09-27 at 2 50 46 PM
Screenshot 2023-09-27 at 2 51 37 PM


**Including Tax:**
Screenshot 2023-09-27 at 2 53 10 PM

Screenshot 2023-09-27 at 2 52 03 PM

As builds are green. Hence, moving this PR to **Merge in progress**

**Pull request merged**